File: DataStructuresC++/Stack/BalancedBracket.cpp
#include <iostream>
using namespace std;
#include <stack>
#include <stddef.h>

// Complete the isBalanced function below.
string isBalanced(string s) {
    bool test = true;
    stack <char>q;
    int a = s.length();
    for (int i = 0; i < a; i ++){
        if ((s[i] == '(') ||(s[i] == '{')  || (s[i] == '['))       {
            q.push(s[i]);  
        }
               
        else if (!q.empty()) {
            if ((s[i] == ')' && q.top() == '(') || (s[i] == ']' && q.top() == '[') || (s[i] == '}' && q.top() == '{')) {
                q.pop();
            }
            else{
                test = false;
            }
                
        }

        else {
            test = false;
        }
             
    }       
    if (q.empty() && test){
        return "YES";
    }
        
    else{
        return "NO";
    }
      

}

File: Data Structures Problems(python)/Stack/BalancedBracket.py
#!/bin/python3

import math
import os
import random
import re
import sys
from queue import LifoQueue
# Complete the isBalanced function below.
def isBalanced(s):
    test = True
    result =""
    q = LifoQueue()
    a = len(s)
    for i in range(a):
        if s[i] == "(" or s[i] == "[" or s[i] == "{":
            q.put(s[i])        
        elif not q.empty():
            top = q.get()
            if (s[i] == ")" and top == "(") or (s[i] == "}" and top == "{") or (s[i] == "]" and top == "[") :
                pass
            else:
                test = False
        else:
            test = False
            
    if q.empty() and test:
        return "YES"
    else:
        return "NO"
if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        s = input()

        result = isBalanced(s)

        fptr.write(result + '\n')

    fptr.close()
